scripts/tools/metricTracer.yml

```diff
@@ -0,0 +1,261 @@
+      name: hMetricTracer
+      desc: "[in] handle of the metric tracer"
+    - type: $x_bool_t
+      name: synchronous
+      desc: |
+            [in] request synchronous behavior. Confirmation of successful asynchronous operation is done by calling $tMetricTracerReadDataExp()
+            and checking the return status: $X_RESULT_SUCCESS will be returned when the tracer is active or when it is inactive but still has data. 
+            $X_RESULT_NOT_READY will be returned when the tracer is inactive and has no more data to be retrieved.
+--- #--------------------------------------------------------------------------
+type: function
+desc: "Read data from the metric tracer"
+version: "1.10"
+class: $tMetricTracer
+name: ReadDataExp
+details:
+    - "The application must **not** call this function from simultaneous threads with the same metric tracer handle."
+    - "Data can be retrieved after tracer is disabled. When buffers are drained $X_RESULT_NOT_READY will be returned"
+params:
+    - type: "$t_metric_tracer_exp_handle_t"
+      name: hMetricTracer
+      desc: "[in] handle of the metric tracer"
+    - type: size_t*
+      name: pRawDataSize
+      desc: |
+            [in,out] pointer to size in bytes of raw data requested to read.
+            if size is zero, then the driver will update the value with the total size in bytes needed for all data available.
+            if size is non-zero, then driver will only retrieve that amount of data. 
+            if size is larger than size needed for all data, then driver will update the value with the actual size needed.
+    - type: uint8_t*
+      name: pRawData
+      desc: "[in,out][optional][range(0, *pRawDataSize)] buffer containing tracer data in raw format"
+returns:
+    - $X_RESULT_WARNING_DROPPED_DATA:
+        - "Metric tracer data may have been dropped."
+    - $X_RESULT_NOT_READY: 
+        - "Metric tracer is disabled and no data is available to read."
+--- #--------------------------------------------------------------------------
+type: function
+desc: "Create a metric decoder for a given metric tracer."
+version: "1.10"
+class: $tMetricDecoder
+name: CreateExp
+params:
+    - type: "$t_metric_tracer_exp_handle_t"
+      name: hMetricTracer
+      desc: "[in] handle of the metric tracer"
+    - type: $t_metric_decoder_exp_handle_t*
+      name: phMetricDecoder 
+      desc: "[out] handle of the metric decoder object"
+--- #--------------------------------------------------------------------------
+type: function
+desc: "Destroy a metric decoder."
+version: "1.10"
+class: $tMetricDecoder
+name: DestroyExp
+params:
+    - type: $t_metric_decoder_exp_handle_t
+      name: phMetricDecoder 
+      desc: "[in] handle of the metric decoder object"
+--- #--------------------------------------------------------------------------
+type: function
+desc: |
+      Return the list of the decodable metrics that can be collected in the tracer for the which the metric decoder 
```

Comment:
question here - why would we want to enumerate decoder handles which may not have a list of "decodable metrics"? i.e, the metrics decoder handles that are enumerated should already be valid in the sense that it is "decodable" right?

maybe I'm missing the intention, but should we reqord for clarity or rename API for clarity?

Reply:
the sequence is: 

1. Crate a tracer.
2. Create a decoder for that tracer if you want to decode. 
3. Get the list of decodable metrics from the decoder. Such that when you decode you know what to map each event to. 

I understand the wording is a bit twisted. Maybe it clearer saying  :

"Get the list of decodable metrics from the decoder"

My intention was to say the the decoder is associated with the tracer, so the list of decodable metrics come from the tracer. I can add this in the details